x/docs/internal/keeper/keeper_test.go

```diff
@@ -451,7 +520,107 @@ func TestKeeper_DocumentsIterator_ExistingList(t *testing.T) {
 // --- Document receipts
 // ----------------------------------
 
-func TestKeeper_SaveDocumentReceipt_EmptyList(t *testing.T) {
+func TestKeeper_SaveDocument(t *testing.T) {
+	tests := []struct {
+		name          string
+		empty         bool
```

Comment:
What does "empty" mean? Empty receipts list?

Reply:
empty set true means emptylist in testing

Reply:
Uhm, instead of grouping most of the logic in separate if/else statement, I believe you can extract a common flow for all test cases and use small if's to differentiate various edge cases (list must be empty at the beginning, uuid must be different...).